awscli/customizations/cloudformation/deployer.py

```diff
@@ -106,6 +108,26 @@ def create_changeset(self, stack_name, cfn_template,
             'Capabilities': capabilities,
             'Description': description,
         }
+
+        # CloudFormation requires template bodies to be less than 51200 bytes
+        # if they are deployed from a file. It supports much larger templates
+        # if the templates exist in s3. If this is a 'big' template, throw it
+        # into s3 first.
+        cfn_template_length = len(cfn_template.encode('utf-8'))
```

Comment:
I've modified the logic slightly - this might have issues if CloudFormation ever supports localized templates. Instead, I've moved this out to `deploy.py` and check the actual file of the template before loading it. 

I've also added appropriate unit tests for this condition.